src/main/java/org/spongepowered/api/text/Texts.java

```diff
@@ -234,6 +251,63 @@ public static Text of(Object... objects) {
             }
         }
 
+        if (builder.children.size() == 1) {
+            // Single content, reduce Text depth
+            return builder.children.get(0);
+        }
+        return builder.build();
+    }
+
+    /**
+     * Builds a {@link Text} from a given array of content objects. This method
+     * assumes, that all elements of the array are content and not formatting.
+     *
+     * <p>This will create the correct {@link Text} instance if the input object
+     * is the input for one of the {@link Text} types or convert the object to a
+     * string otherwise.</p>
+     *
+     * @param separator The separator to be inserted between elements
+     * @param objects The object array
+     * @return The built text object
+     */
+    public static Text joinOf(Text separator, Object... objects) {
+        TextBuilder builder = builder();
+        boolean first = true;
+
+        for (Object part : objects) {
+            if (!first) {
+                builder.append(separator);
+            }
+            first = false;
+            builder.append(Texts.of(part));
```

Comment:
Yeah, I know what you mean, but your usecase is very unlikely (IMO).
Because you can't distinguish each team, because its flatten away.

````
The teams are Player1, Player2, Player3, Player4, Player5, Player6, Player7, Player8, Player9
````

Which player belongs to which team?
So you would probably send a message/line per Team or at a separator between the teams:
````java
List<Text> texts = List.newList();
text.add(Texts.of("The teams are "));
for (Set<Player> members : teams) {
    text.add(Texts.join(Texts.of(", "), members));
}
player.sendMessage(Texts.join(Texts.of("\n"), texts));
````
Which results in:
````
The teams are 
Team1: Player1, Player2, Player3
Team2: Player4, Player5, Player6
Team3: Player7, Player8, Player9
````
Do you actually want to use that method or do you just think it could be useful?
In the first case I would add it, until discussed away.
The the second case I would wait until we decided on its usefulness.